runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,321 @@
+		config.timeOfDeath(Instant.now());
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane()));
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			worldMapPointManager.add(new DeathWorldMapPoint(new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane())));
+		}
+
+		resetInfobox();
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() == ChatMessageType.SERVER)
+		{
+			if (event.getMessage().equals("Oh dear, you are dead!"))
+			{
+				hasRespawned = true;
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.hasDied() || !hasRespawned || (client.getWorld() != config.deathWorld()))
+		{
+			return;
+		}
+
+		// Check if the player is at their death location, or timer has passed
+		WorldPoint deathPoint = new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane());
+		if (deathPoint.equals(client.getLocalPlayer().getWorldLocation()) || (deathTimer != null && deathTimer.cull()))
+		{
+			client.clearHintArrow();
+
+			if (deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+
+			resetDeath();
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("deathIndicator"))
+		{
+			if (!config.showDeathHintArrow() && config.hasDied())
+			{
+				client.clearHintArrow();
+			}
+
+			if (!config.showDeathInfoBox() && deathTimer != null)
+			{
+				infoBoxManager.removeInfoBox(deathTimer);
+				deathTimer = null;
+			}
+
+			if (!config.showDeathOnWorldMap())
+			{
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+
+			if (!config.hasDied())
+			{
+				client.clearHintArrow();
+
+				resetInfobox();
+
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (!config.hasDied())
+		{
+			return;
+		}
+
+		if (event.getGameState() == GameState.LOGGED_IN)
+		{
+			if (client.getWorld() == config.deathWorld())
+			{
+				WorldPoint deathPoint = new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane());
+
+				if (config.showDeathHintArrow())
+				{
+					client.setHintArrow(deathPoint);
+				}
+
+				if (config.showDeathOnWorldMap())
+				{
+					worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+					worldMapPointManager.add(new DeathWorldMapPoint(deathPoint));
+				}
+			}
+			else
+			{
+				worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			}
+		}
+	}
+
+	private void resetDeath()
+	{
+		config.hasDied(false);
```

Comment:
Instead of using hasDied, just null the time of death here, and remove the entire hasDied boolean and use time of death checks instead.

Reply:
Fixed.